pkg/validation/utils.go

```diff
@@ -21,11 +21,9 @@ func getCommandsMap(commands []v1alpha2.Command) map[string]v1alpha2.Command {
 }
 
 // isInt returns true if the string is an integer, false otherwise
-func isInt(str string) bool {
-	if _, err := strconv.Atoi(str); err == nil {
-		return true
-	}
-	return false
+func isInt(str string) (bool, error) {
+	match, err := regexp.MatchString("^[0-9]+$", str)
```

Comment:
Two minor issues here:
1. Regex won't match `10-10`; if you want to check that at least one letter exists, just check `"[a-zA-Z]"`
2. Avoid returning an error by using 
    ```
    var alphaRegexp = regexp.MustCompile(`[a-zA-Z]`)
    ```
    at the top of the file.

Reply:
+1 on point 2, thx

On point 1, I just chatted with Stephanie who wrote the initial doc for validation and it seems to her `10-10` would be a valid id. I just tried it out and it looks Kube seems to be happy with it. 
Looks like we only want to restrict `1234`. Do you agree or have a counter argument?

Reply:
If `10-10` is valid but `1234` is not, I wonder where the restriction is coming from. For what it's worth, `1234` is a valid identifier in k8s as well (you have to quote it in yaml otherwise it's interpreted as an integer).

The only checks for "not all numbers" I know of are
1. Places that use RFC1035 DNS labels (must start with an alphabetic character)
2. Port names, for which `10-10` is [invalid](https://github.com/kubernetes/apimachinery/blob/ee96543cc6730a24af1c7323dac47e52c340ba39/pkg/util/validation/validation.go#L317-L345)

In this case, what is the `isInt` check being used for specifically? I tried applying a DevWorkspace with ids like `"1234"` littered throughout and I didn't run into any issues -- the DevWorkspace operator correctly named a container `1234`.

Reply:
Actually I wasnt able to create a svc with `1234`/`12-34`. Not sure what resource the DevWorkspace was creating 🤔. I even quoted it in yaml :
```
The Service "1234" is invalid: metadata.name: Invalid value: "1234": a DNS-1035 label must consist of lower case alphanumeric characters or '-', start with an alphabetic character, and end with an alphanumeric character (e.g. 'my-name',  or 'abc-123', regex used for validation is '[a-z]([-a-z0-9]*[a-z0-9])?')
```

But anyhow, I think we can remove this restriction. I confirmed and it seems like port name cannot take either `1234`/`12-34` like you suggested and linked. And the library generator is [already handling this](https://github.com/devfile/library/blob/741af7b5cad69d4962c8ea1b374d9c80014eb6b9/pkg/devfile/generator/utils.go#L46) for container port names from devfile. 

Command IDs are not representative of any Kube resource and this just leaves us with Component name. I was able to create a pod with container named `"1234"`/`"12-34"`.

WDYT?

Reply:
Makes sense to me, it seems like `isInt` is unnecessary (I believe the `1234` svc name is covered by in-schema regexes)

Reply:
Updated! I have also addressed the remaining unresolved conversations.